apps/token-manager/app/src/components/Panels/AssignVotePanelContent.js

```diff
@@ -2,89 +2,132 @@ import React from 'react'
 import styled from 'styled-components'
 import { Button, Field, IconCross, Text, TextInput } from '@aragon/ui'
 import { addressPattern, isAddress } from '../../web3-utils'
-import { fromDecimals, toDecimals } from '../../utils'
+import { fromDecimals, toDecimals, formatBalance } from '../../utils'
 
 // Any more and the number input field starts to put numbers in scientific notation
 const MAX_INPUT_DECIMAL_BASE = 6
 
 const initialState = {
-  amount: '',
   mode: 'assign',
-  holder: {
+  holderField: {
     error: false,
     value: '',
   },
+  amountField: {
+    error: false,
+    value: '',
+    max: '',
+  },
 }
 
 class AssignVotePanelContent extends React.Component {
   static defaultProps = {
     onUpdateTokens: () => {},
   }
-  constructor(props) {
-    super(props)
-    this.state = {
-      ...initialState,
-      ...props.holder,
-    }
+  state = {
+    ...initialState,
   }
-  componentWillReceiveProps({ opened, holder = '', mode }) {
+  componentWillReceiveProps({ opened, mode, holderAddress }) {
     if (opened && !this.props.opened) {
       // setTimeout is needed as a small hack to wait until the input is
       // on-screen before we call focus
       this.holderInput && setTimeout(() => this.holderInput.focus(), 0)
 
-      // Holder override passed in from props
-      this.setState({
-        holder: {
-          ...initialState.holder,
-          value: holder,
-        },
-      })
+      // Upadte holder address from the props
+      this.updateHolderAddress(mode, holderAddress)
     }
 
     // Finished closing the panel, its state can be reset
     if (!opened && this.props.opened) {
       this.setState({ ...initialState })
     }
   }
-  handleAmountChange = event => {
-    this.setState({ amount: event.target.value })
+  filteredHolderAddress() {
```

Comment:
Should these new functions all use the assignment notation since they access `this`?

Reply:
I usually only do it when the method need to be called from somewhere else, otherwise I try to keep it as a method. That way it’s only declared once on the prototype, rather than being attached to every instance 🏋️‍♂️